tests/unittest/_torch/attention/sparse/dsa/test_dsa_sparse_mla.py

```
@@ -0,0 +1,1007 @@
+    )
+    AttentionCls = get_attention_backend(backend_name, sparse_config)
+    # When rope_append is False, [448: 512) are used for qk_rope_head_dim
+    kv_lora_rank = kv_lora_rank - qk_rope_head_dim if not rope_append else kv_lora_rank
+    head_dim = kv_lora_rank + qk_rope_head_dim
+
+    # Set seed for reproducibility.
+    torch.manual_seed(123)
+
+    # Create inputs
+    inputs_per_layer = []
+    for layer_idx in range(num_layers):
```

Ruff (0.15.18)

[warning] Loop control variable `layer_idx` not used within loop body

(B007)